### PR TITLE
Pull arch from payload

### PR DESCRIPTION
**Pull request by jottofar**
and use when querying OSUS. Per https://issues.redhat.com/browse/OTA-655, heterogeneous cluster payloads will contain metadata key "architecture" set to "multi". Homogeneous cluster payloads are unchanged and will not contain the metadata key "architecture" in which case CVO will set the architecture using runtime.GOARCH.

### CODE REVIEW

File: pkg/cvo/cvo.go
Comment (wking) [inline pkg/cvo/cvo.go:97]: > ... if the release image declares 'release.openshift.io/architecture: multi'.

right?  Because you can be, and ask for, heterogeneous release images without actually having heterogeneous nodes.  And the homogeneous -> heterogeneous transition will (eventually) be:

1. Running a homogeneous release on a homogeneous cluster, asking for homogeneous update recommendations.
2. Running a homogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
3. Updating to a heterogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
4. Running a heterogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
5. Running a heterogeneous release on a heterogenous cluster, asking for heterogeneous update recommendations.

File: pkg/payload/payload.go
Comment (wking) [inline pkg/payload/payload.go:443]: nit: at the moment, the `==` seems straightforward enough that I'd rather inline it and drop `isHeterogeneousArchitecture`.  But I don't feel all that strongly about it, so feel free to ignore me if you prefer it the way you have it.

File: pkg/cvo/cvo.go
Comment (jottofar) [reply pkg/cvo/cvo.go:97]: Yeah, agree, but that's the "to be" logic. Implementation of this Jira is just a baby step so I made the comment only reflect what this code does now. This value simply gets to set to either "mullti" if the payload contains it or GOARCH. We can expand when we add more logic.

File: pkg/payload/payload.go
Comment (jottofar) [reply pkg/payload/payload.go:443]: Prefer to leave it and just move on since it has to be touched again to fully implement heter.

**Review by wking — APPROVED** [review]
/lgtm

**Comment by LalatenduMohanty** [comment]
Still working on the review.

File: pkg/cvo/cvo.go
Comment (LalatenduMohanty) [inline pkg/cvo/cvo.go:97]: Can we add about the possible value of architecture when cluster is heterogeneous i.e. `multi` and for others it would be `GOARCH`.

**Comment by LalatenduMohanty** [comment]
Added some comments which are nit picks and does not need to be fixed before the merge.

File: pkg/cvo/availableupdates.go
Comment (wking) [reply pkg/cvo/availableupdates.go:170]: What would we log?  Something like "bumping LastSyncOrConfigChange because..." and then "upstream changed from A to B", "channel changed from A to B", or "we successfully synchronized available updates"?

File: pkg/cvo/cvo.go
Comment (wking) [reply pkg/cvo/cvo.go:97]: if we want to doc this, I think we should just say "to match the Cincinnati API's 'arch' property" or something, and then update [this][1] in  openshift/cincinnati#711 to cover `multi` as well.

[1]: https://github.com/openshift/cincinnati/blob/master/docs/design/openshift.md#request

File: pkg/cvo/availableupdates.go
Comment (wking) [inline pkg/cvo/availableupdates.go:229]: Not sure why yet, but this is [tripping e2e-agnostic like][1]:

```
: [sig-instrumentation][Late] Alerts shouldn't report any unexpected alerts in firing or pending state [Suite:openshift/conformance/parallel]
--
Run #0: Failed 5s
{  fail [github.com/onsi/ginkgo@v4.7.0-origin.0+incompatible/internal/leafnodes/runner.go:113]: Jul 28 15:51:34.182: Unexpected alerts fired or pending after the test run:  alert CannotRetrieveUpdates fired for 313 seconds with labels: {endpoint="metrics", instance="10.0.0.8:9099", job="cluster-version-operator", namespace="openshift-cluster-version", pod="cluster-version-operator-7674b7b444-f7rn9", service="cluster-version-operator", severity="warning"} ...
```

and:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432/artifacts/e2e-agnostic/gather-extra/artifacts/clusterversion.json | jq -r '.items[].status.conditions[] | select(.type == "RetrievedUpdates")'
{
  "lastTransitionTime": "2022-07-28T14:21:43Z",
  "message": "The set of architectures has not been configured.",
  "reason": "NoArchitecture",
  "status": "False",
  "type": "RetrievedUpdates"
}
```

/hold

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432

Comment (wking) [reply pkg/cvo/availableupdates.go:229]:
```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432/artifacts/e2e-agnostic/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-7674b7b444-f7rn9_cluster-version-operator.log | grep 'Payload loaded from'
I0728 14:46:02.143948       1 sync_worker.go:385] Payload loaded from registry.build02.ci.openshift.org/ci-op-7ipcn54g/release@sha256:70dd7baaa8c8550e19dddbe79d75a9d78bc37855974649b8292da0e1dccb4d56 with hash oiJMeRYBbXY=, architecture 
```

not sure if that's relevant, but sounds suspicious.

File: pkg/cvo/availableupdates.go
Comment (LalatenduMohanty) [reply pkg/cvo/availableupdates.go:170]: Never mind. I misunderstood the purpose of this

File: pkg/cvo/cvo.go
Comment (LalatenduMohanty) [reply pkg/cvo/cvo.go:97]: I am fine with that as long as we can find out from the doc what values can be used for it.

Comment (jottofar) [reply pkg/cvo/cvo.go]: Updated comment

File: pkg/payload/payload.go
Comment (jottofar) [reply pkg/payload/payload.go]: done

File: pkg/cvo/cvo.go
Comment (jottofar) [reply pkg/cvo/cvo.go]: Updated comment

File: pkg/cvo/availableupdates.go
Comment (jottofar) [reply pkg/cvo/availableupdates.go:229]: Needed call to `SetArchitecture` in `InitializeFromPayload`

**Review by wking — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
[`TestCVO_UpgradeFailedPayloadLoadWithCapsChanges`][1] is a frequent race recently.  We should fix it, but not in this PR.

/override ci/prow/unit

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-unit/1552761191300861952

**Review by wking — APPROVED** [review]
/lgtm

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Comment by jottofar** [comment]
TestCVO_UpgradeFailedPayloadLoadWithCapsChanges race, unrelated to this PR

/override ci/prow/unit

**Pull request merged**
